### ISSUE TITLE
It is currently impossible to test external package loading from within the editor

### ISSUE THREAD

**Issue by EIREXE** [godotengine/godot#16798]
**Godot version:** 3.0 Release

**OS/device including version:** Windows

**Issue description:**
When loading resource packs using ProjectSettings.load_resource_pack one would assume the content from the resource pack is added to the virtual filesystem, this does work but it also breaks other things, I now present three cases.

In each of the following cases the pck has a folder structure of Content/ModTest and two files in that folder, modtest.png and project.json, modtest.png is a picture of Godette and project.json is some text data.

The game itself has two folders inside the Content folder, BaseContent and TestContent.

**1st case: Directly loading resource using load():** This works exactly as expected, no issues about it.

![imagen](https://user-images.githubusercontent.com/3101690/36355308-844cd3b2-14e1-11e8-94af-a0804ffc8d45.png)

**2nd case: Iterating a directory using Directory.list_dir**: It doesn't work, only the files and folders from the newly loaded pck file seem to be found, this is my method:
```

func test_pck_issue():
	var dir = Directory.new()
	if dir.open("res://Content") == OK:
		dir.list_dir_begin(true)
		var directory_name = dir.get_next()
		while (directory_name != ""):
			if dir.current_is_dir():
				print(directory_name)
				file_name = dir.get_next()

```

In this case only ModTest is found, this doesn't seem right to me.

Output when ProjectSettings.load_resource_pack is used:
`ModTest`

Output when it's not used:
```
BaseContent
TestContent
```

**3rd case: trying to find files using the file browser**: This only seems to find the game's base content, but none of the newly loaded pck content.

![imagen](https://user-images.githubusercontent.com/3101690/36355256-deb82e60-14e0-11e8-9888-7440243677ae.png)

I have also tried removing the project configuration from the package, it doesn't change anything.

**Comment by EIREXE** [comment]
PetePete1984 said in the discord that while load_resource_pack is undocumented it uses PackedData::add_path which is supposed to be additive, so it seems like it is indeed a bug.

**Comment by EIREXE** [comment]
I figured out what's causing this, the problem is that the ProjectSettings singleton sets DirAccess::make_default<DirAccessPack>(DirAccess::ACCESS_RESOURCES);

Because it makes all newly created DirAccess instances use DirAccessPack it only loads from resource packs, and since we are running in the editor we don't really have a resource pack.

load() doesn't use DirAccess, that's why it has no issues loading from both

I am not sure how this can be tackled, if it even can, if I knew I would fix it myself.

**Comment by PetePete1984** [comment]
Might be useful to add that it works in export mode, where all access to files is through Packs; where it breaks is if you try to use both Packs and the filesystem, ie. in the editor.

**Comment by EIREXE** [comment]
I originally proposed to use a hybrid of all three if we are using a mix, but I am not sure if there's a better way.

**Comment by eon-s** [comment]
If `load` work with pck on debug mode it can be used for testing while the other is used on release with conditionals, or make `load_resource_pack` _optionally_ behave differently on debug mode.

**Comment by EIREXE** [comment]
@eon-s `load` actually works with multiple sources regardless of being in debug or in release mode, because `load` does not use `DirAccess`, it does it directly instead, so even if you mix fileystem resources and packages it still works, as long as you supply it the proper path.

The problem comes when mixing them both, once a package is loaded `Directory` can't find any resource that was outside the pck because `Directory` stops using `DirAccess `and starts using `DirAccessPack`, which only looks at files in pcks.

**Comment by EIREXE** [comment]
I have changed the title to a more descriptive one

**Comment by Calinou** [comment]
@EIREXE Can you still reproduce this bug in Godot 3.2.1 or [3.2.2beta4](https://godotengine.org/article/dev-snapshot-godot-3-2-2-beta-4)?

**Comment by EIREXE** [comment]
Yes, this is still an issue, the engine's design still has this problem.

**Comment by cloewen8** [comment]
Still a problem with Godot 3.2.3 and 3.2.2 from what I found. I made a minimal test case to report it as bug, before I found this issue.
```gd
func _ready():
	var packer = PCKPacker.new();
	packer.pck_start("res://pack.pck");
	packer.flush(true);
	ProjectSettings.load_resource_pack("pack.pck")
	var dir = Directory.new()
	assert(dir.open("res://dir") == OK, "Could not open the folder 'dir'.")
```
